engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -163,6 +163,7 @@ public StramClient(Configuration conf, LogicalPlan dag) throws Exception
   {
     this.conf = conf;
     this.dag = dag;
+    dag.visitDAG();
```

Comment:
Would it be useful to also allow visitor access to the dag before the properties are set.